pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -227,10 +228,15 @@ def __init__(
         leave: bool = False,
         theme: RichProgressBarTheme = RichProgressBarTheme(),
     ) -> None:
-        if not _RICH_AVAILABLE:
+        if not _module_available("rich"):
```

Comment:
This could still use `_RICH_AVAILABLE` as the message already suggests using `-U`

See how it's done for the CLI:

https://github.com/PyTorchLightning/pytorch-lightning/blob/5d9df39b1c35629c25ed3faba094a03b07dec15c/pytorch_lightning/utilities/imports.py#L87

https://github.com/PyTorchLightning/pytorch-lightning/blob/5d9df39b1c35629c25ed3faba094a03b07dec15c/pytorch_lightning/utilities/cli.py#L107-L111


There's no need for 2 separate checks in this case

Reply:
Yup, we had it initially for Rich as well. But imo it is better error handling for the user stating what's missing.

Reply:
You can still state it in that case, by keeping only this if but the message in the bottom error.

Reply:
if rich is not installed, it also misses the requirement of `rich >= 10.2.2`